core/templates/dev/head/exploration_editor/RouterServices.js

```diff
@@ -137,34 +158,26 @@ oppia.factory('routerService', [
       },
       navigateToMainTab: function(stateName) {
         _savePendingChanges();
-
         if (_getCurrentStateFromLocationPath() === stateName) {
           return;
         }
 
-        var _actuallyNavigate = function(newStateName) {
-          if (newStateName) {
-            editorContextService.setActiveStateName(newStateName);
-          }
-          $location.path('/gui/' + editorContextService.getActiveStateName());
-          $window.scrollTo(0, 0);
-        };
-
         if (_tabs.active === MAIN_TAB) {
           $('.oppia-editor-cards-container').fadeOut(function() {
-            _actuallyNavigate(stateName);
+            _actuallyNavigate(SLUG_GUI, stateName);
             $rootScope.$apply();
             $timeout(function() {
               $('.oppia-editor-cards-container').fadeIn();
             }, 150);
           });
         } else {
-          _actuallyNavigate(stateName);
+          _actuallyNavigate(SLUG_GUI, stateName);
         }
       },
       navigateToPreviewTab: function() {
-        _savePendingChanges();
-        $location.path('/preview');
+        if (_tabs.active !== PREVIEW_TAB) {
```

Comment:
Oops, no, it wasn't intentional, I'll restore that.  The changes are saved anyway, presumably because of the $rootScope.$broadcast('externalSave') at line 56.  Should that really be a call to _savePendingChanges() too?  Or isn't the try/catch needed in that case?